pkg/admin/admin_config.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package admin
+
+import (
+	"github.com/streamnative/pulsarctl/pkg/pulsar/common"
+
+	"github.com/streamnative/terraform-provider-pulsar/pkg/authentication"
+)
+
+type PulsarAdminConfig struct {
+	Config *common.Config
+}
+
+func (p *PulsarAdminConfig) AuthenticationType() authentication.AuthenticationType {
+	if len(p.Config.IssuerEndpoint) > 0 && len(p.Config.ClientID) > 0 &&
+		len(p.Config.Audience) > 0 && len(p.Config.KeyFile) > 0 {
```

Comment:
This field is optional, do we need to check it? for example on azure AD OAuth2 this field is not needed